examples/vision/transformer_in_transformer.py

```diff
@@ -0,0 +1,328 @@
+    x = layers.Dense(embedding_dim)(x)
+    x = layers.Dropout(drop_rate)(x)
+    return x
+
+
+"""
+## Implement the TNT block
+"""
+
+
+def transformer_in_transformer_block(
+    pixel_embedding,
+    patch_embedding,
+    out_embedding_dim,
+    in_embedding_dim,
+    num_pixel,
+    out_num_heads=8,
+    in_num_heads=4,
+    mlp_ratio=4,
+    attention_dropout=0,
+    projection_dropout=0,
+):
+    # inner transformer block
+    residual_in_1 = pixel_embedding
+    pixel_embedding = layers.LayerNormalization(epsilon=1e-5)(pixel_embedding)
+    pixel_embedding = layers.MultiHeadAttention(
+        num_heads=in_num_heads, key_dim=in_embedding_dim, dropout=attention_dropout
+    )(pixel_embedding, pixel_embedding)
+    pixel_embedding = layers.add([pixel_embedding, residual_in_1])
+    residual_in_2 = pixel_embedding
+    pixel_embedding = layers.LayerNormalization(epsilon=1e-5)(pixel_embedding)
+    pixel_embedding = mlp(
+        pixel_embedding, in_embedding_dim * mlp_ratio, in_embedding_dim
+    )
+    pixel_embedding = layers.add([pixel_embedding, residual_in_2])
+
+    # outer transformer block
+    _, num_patches, channel = patch_embedding.shape
+    # fusion local and global information
+    fusion_embedding = tf.reshape(
+        pixel_embedding, shape=(-1, num_patches, in_embedding_dim * num_pixel)
+    )
+    fusion_embedding = layers.LayerNormalization(epsilon=1e-5)(fusion_embedding)
+    fusion_embedding = layers.Dense(out_embedding_dim)(fusion_embedding)
+    patch_embedding = layers.add([patch_embedding, fusion_embedding])
+    residual_out_1 = patch_embedding
+    patch_embedding = layers.LayerNormalization(epsilon=1e-5)(patch_embedding)
+    patch_embedding = layers.MultiHeadAttention(
+        num_heads=out_num_heads, key_dim=out_embedding_dim, dropout=attention_dropout
+    )(patch_embedding, patch_embedding)
+    patch_embedding = layers.add([patch_embedding, residual_out_1])
+    residual_out_2 = patch_embedding
+    patch_embedding = layers.LayerNormalization(epsilon=1e-5)(patch_embedding)
+    patch_embedding = mlp(
+        patch_embedding, out_embedding_dim * mlp_ratio, out_embedding_dim
+    )
+    patch_embedding = layers.add([patch_embedding, residual_out_2])
+    return pixel_embedding, patch_embedding
+
+
+"""
+## Implement the TNT model
+The TNT model consists of multiple TNT blocks.
+In the TNT block, there are two transformer blocks where
+the outer transformer block models the global relation among patch embeddings,
+and the inner one extracts local structure information of pixel embeddings.
+The local information is added on the patch
+embedding by linearly projecting the pixel embeddings into the space of patch embedding.
+Patch-level and pixel-level position embeddings are introduced in order to
+retain spatial information.In orginal paper, the authors use the class token for
+classification.
+We use the `layers.GlobalAvgPool1D` to fusion patch information.
+"""
+
+
+def get_model(
+    image_size=96,
```

Comment:
You are repeating a number of constants here that were defaulted in your helper functions. Either place is fine, but don't duplicate these numbers. It makes it hard what to tell what was used (e.g. outer_block_num_heads is 4 here, and default to out_num_heads is 8 in transformer_in_transformer_block).